python/cudf/cudf/core/frame.py

```diff
@@ -582,16 +574,35 @@ def equals(self, other, **kwargs):
         else:
             return self._index.equals(other._index)
 
-    def _explode(self, explode_column_num: int, index: Optional[cudf.Index]):
-        if index is not None:
-            explode_column_num += index.nlevels
-        res_tbl = libcudf.lists.explode_outer(
-            cudf._lib.table.Table(self._data, index=index), explode_column_num
-        )
+    def _explode(self, explode_column: Any, ignore_index: bool):
+        """Helper function for `explode` in Series and Dataframe.
+        if the designated column to explode is non-nested, a copy
+        of the frame is returned. Otherwise, if ignore_index is
+        set, the original index is not exploded and will use
+        a `RangeIndex` instead.
+        """
+        if not is_list_dtype(self._data[explode_column].dtype):
+            copy = self.copy()
+            if ignore_index:
+                copy._index = cudf.RangeIndex(copy._num_rows)
```

Comment:
If we're ignoring the index then we're making a deep copy of the index unnecessarily here before getting rid of it with a new `RangeIndex`.